src/ripple/app/ledger/Consensus.h

```diff
@@ -0,0 +1,82 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_APP_LEDGER_IMPL_CONSENSUS_H_INCLUDED
+#define RIPPLE_APP_LEDGER_IMPL_CONSENSUS_H_INCLUDED
+
+#include <ripple/app/ledger/LedgerConsensus.h>
+#include <ripple/app/misc/FeeVote.h>
+#include <ripple/app/misc/NetworkOPs.h>
+#include <ripple/app/tx/InboundTransactions.h>
+#include <ripple/app/tx/LocalTxs.h>
+
+#include <beast/cxx14/memory.h> // <memory>
+
+namespace ripple {
+
+/** Implements the consensus process and provides inter-round state. */
+class Consensus
```

Comment:
I'm going to pile on, but my main objection is the fact that the `ConsensusImp` is passed to the `LedgerConsensusImp` so that it can call these semi-private methods. My understanding is that these `Imp` classes are supposed to be practically private. While I'm sure there are counter examples, I don't think it's a good idea to introduce new one.

Also, there is nearly circular dependency between ConsensusImp.h and LedgerConsensusImp.h, because of the call to `make_LedgerConsensus`. 

It just seems that these classes could be organized better.